lib/array_equals.rb

```diff
@@ -1,5 +1,20 @@
 # Determines if the two input arrays have the same count of elements
 # and the same integer values in the same exact order
 def array_equals(array1, array2)
-  raise NotImplementedError
+  if array1 == nil || array2 == nil
+    return array1 == array2
+  end
+  if array1.length == 0 || array2.length == 0
+    return array1 == array2
```

Comment:
Actually in this case you could just return true.  I think it's possible to have 2 separate empty arrays that are not equal with the ==.